src/node_buffer.cc

```diff
@@ -1142,6 +1142,143 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
                                 : -1);
 }
 
+void NoopCallback(char* data, void* hint) {
+  // Intentional no-op
+}
+
+void Split(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  THROW_AND_RETURN_UNLESS_BUFFER(env, args[0]);
+  SPREAD_ARG(args[0], js_buf);
+  SPREAD_ARG(args[1], js_sep_buf);
+
+  std::vector<size_t> index_arr;
+  Local<Object> obj;
+  Persistent<Object> persistent(env->isolate(), obj);
+  Local<v8::Array> res_buffers;
+  size_t cursor = 0;
+  size_t len = 0;
+
+  // Rationale: Find all occurences of the buffer bytes, store those index and
+  // copy bytes from index bounds to new array of buffers
+
+  // REVIEW(eljefedelrodeodeljefe): copy first necessary?
+  char* buf = new char[js_buf_length];
```

Comment:
use `malloc()`, check if `nullptr`. if so then throw an error that data couldn't be allocated. this will abort if that's the case.